libtiledbsoma/src/soma/soma_geometry_column.cc

```diff
@@ -0,0 +1,365 @@
+
+    for (int64_t j = 0; j < spatial_schema->n_children; ++j) {
+        dims.push_back(ArrowAdapter::tiledb_dimension_from_arrow_schema(
+            ctx,
+            spatial_schema->children[j],
+            spatial_array->children[j],
+            soma_type,
+            type_metadata,
+            SOMA_GEOMETRY_DIMENSION_PREFIX,
+            "__max",
+            platform_config));
+    }
+
+    auto attribute = ArrowAdapter::tiledb_attribute_from_arrow_schema(
+        ctx, schema, type_metadata, platform_config);
+
+    return std::make_shared<SOMAGeometryColumn>(
+        SOMAGeometryColumn(dims, attribute.first));
+}
+
+void SOMAGeometryColumn::_set_dim_points(
+    const std::unique_ptr<ManagedQuery>& query,
+    const SOMAContext& ctx,
+    const std::any& points) const {
+    std::vector<std::pair<double_t, double_t>>
+        transformed_points = _transform_points(
+            std::any_cast<std::span<const std::vector<double_t>>>(points));
+
+    auto domain_limits = _limits(ctx, *query->schema());
+
+    // Create a range object and reuse if for all dimensions
+    std::vector<std::pair<double_t, double_t>> range(1);
+    size_t dimensionality = dimensions.size() / 2;
+
+    for (size_t i = 0; i < transformed_points.size(); ++i) {
+        range[0] = std::make_pair(
+            domain_limits[i].first,
+            std::min(transformed_points[i].second, domain_limits[i].second));
+        query->select_ranges(dimensions[i].name(), range);
+
+        range[0] = std::make_pair(
+            std::max(transformed_points[i].first, domain_limits[i].first),
+            domain_limits[i].second);
+        query->select_ranges(dimensions[i + dimensionality].name(), range);
+    }
+}
+
+void SOMAGeometryColumn::_set_dim_ranges(
+    const std::unique_ptr<ManagedQuery>& query,
+    const SOMAContext& ctx,
+    const std::any& ranges) const {
+    std::vector<std::pair<double_t, double_t>>
+        transformed_ranges = _transform_ranges(
+            std::any_cast<std::vector<
+                std::pair<std::vector<double_t>, std::vector<double_t>>>>(
+                ranges));
+
+    auto domain_limits = _limits(ctx, *query->schema());
+
+    // Create a range object and reuse if for all dimensions
+    std::vector<std::pair<double_t, double_t>> range(1);
+    size_t dimensionality = dimensions.size() / 2;
+
+    for (size_t i = 0; i < transformed_ranges.size(); ++i) {
+        range[0] = std::make_pair(
+            domain_limits[i].first,
+            std::min(transformed_ranges[i].second, domain_limits[i].second));
+        query->select_ranges(dimensions[i].name(), range);
+
+        range[0] = std::make_pair(
+            std::max(transformed_ranges[i].first, domain_limits[i].first),
+            domain_limits[i].second);
+        query->select_ranges(dimensions[i + dimensionality].name(), range);
+    }
+}
+
+void SOMAGeometryColumn::_set_current_domain_slot(
+    NDRectangle& rectangle, std::span<const std::any> domain) const {
```

Comment:
As above: we must not use the plain variable name `domain` without qualification, as it is ambiguous and error-prone.